src/firecracker/build.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::process::Command;
+
+// this build script is called on every firecracker package build,
+// embedding the FIRECRACKER_VERSION directly in the resulting binary
+fn main() {
+    let git_version_command = Command::new("git")
+        .args(&["describe", "--dirty"])
+        .output()
+        .unwrap();
+
+    if git_version_command.status.success() {
+        println!(
+            "cargo:rustc-env=FIRECRACKER_VERSION={}",
+            String::from_utf8(git_version_command.stdout)
+                .unwrap()
+                .trim_start_matches('v')
+        );
+    } else {
+        // if this is not a git repo, return the standard cargo package version
+
+        println!(
+            "cargo:rustc-env=FIRECRACKER_VERSION={}",
+            env!("CARGO_PKG_VERSION")
+        );
+    }
+
+    Command::new("touch").args(&["build.rs"]).output().unwrap();
```

Comment:
This line is better suited in `devtool` on the `build` path.

We don't want this mechanism to always re-trigger even in tests or local development (we don't want to wait for repeated unnecessary `release` linking each time we run a `pytest` test).

`devtool build [--release]` can `touch` this and force the binary being rebuilt and having the right `version`.
All other `devtool` paths (`test/shell`) should not _force_ rebuilding the binary.

WDYT?

Reply:
I agree with you, but what if someody does a build using `cargo build` instead of using devtool?
I figured that implementing it this way, this would be devtool-agnostic.
WDYT?

Reply:
Local (non-devtool/non-container) builds should never reach clients, all our automated scripts and runbooks for manual steps work with `devtool`.

I would like to support `cargo build` to _always_ update the right version, but not at the expense on development speed.

Reply:
done. added this `touch` command in devtool on the build path